tests/other/test_dlopen_async.c

```diff
@@ -0,0 +1,28 @@
+#include <emscripten/emscripten.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <stdio.h>
+
+int mydata = 0;
+
+void onsuccess(void *user_data, void *handle) {
+  assert(user_data == &mydata);
+  printf("onsuccess\n");
+  int* foo = (int*)dlsym(handle, "foo");
+  assert(foo);
+  printf("foo = %d\n", *foo);
+  assert(*foo == 42);
+  exit(0);
+}
+
+void onerror(void *user_data) {
+  assert(user_data == &mydata);
+  printf("onerror %s\n", dlerror());
+}
+
+int main() {
+  emscripten_dlopen("libside.so", RTLD_NOW, &mydata, onsuccess, onerror);
+  printf("returning from main\n");
```

Comment:
should we maybe put a call to `dlsym` here and expect that it fails with a useful message?

Reply:
We don't have any kind of handle at this point so what would I pass to dlsym as arg0?